python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -213,3 +221,128 @@ def pairwise_linestring_distance(offsets1, xs1, ys1, offsets2, xs2, ys2):
             )
         }
     )
+
+
+def pairwise_point_linestring_distance(
+    points: GeoSeries, linestrings: GeoSeries
+):
+    """Compute shortest distance between pairs of points and linestrings
+
+    The shortest distance between a pair of (multi)point and (multi)linestring
+    is defined as the shortest distance between every geometry in the
+    multipoint and every geometry in the multilinestring.
```

Comment:
I think we should be explicit about what the geometric test is here.

```suggestion
    is defined as the shortest distance between every point in the
    multipoint and every line segment in the multilinestring.
```